public_html/script.js

```diff
@@ -594,6 +616,29 @@ function refreshSelected() {
                 $('#selected_seen').text(selected.seen.toFixed(1) + 's');
         }
 
+// add the country and flag
```

Comment:
When you initialize the PlaneObject for the first time, stash metadata about the country on the plane object itself (e.g. just storing a reference to the right object would be fine). Then you don't need this lookup and you can just pull the right image path / country name from the metadata without a search.

Also, again, please move the structural bits of the HTML to the static HTML and just update image paths etc rather than creating new DOM nodes on the fly.